src/scoping-shim.js

```diff
@@ -87,29 +87,31 @@ export default class ScopingShim {
     template.extends = typeExtension;
     templateMap[elementName] = template;
     let cssBuild = StyleUtil.getCssBuild(template);
-    let cssText = this._gatherStyles(template);
+    const optimalBuild = StyleUtil.isOptimalCssBuild(cssBuild);
     let info = {
       is: elementName,
       extends: typeExtension,
     };
+    let cssText = this._gatherStyles(template);
     // check if the styling has mixin definitions or uses
     this._ensure();
-    let hasMixins = !StyleUtil.elementHasBuiltCss(template) && detectMixin(cssText);
-    let ast = parse(cssText);
-    // only run the applyshim transforms if there is a mixin involved
-    if (hasMixins && nativeCssVariables && this._applyShim) {
-      this._applyShim['transformRules'](ast, elementName);
+    if (!optimalBuild) {
+      let hasMixins = !StyleUtil.elementHasBuiltCss(template) && detectMixin(cssText);
+      let ast = parse(cssText);
+      // only run the applyshim transforms if there is a mixin involved
+      if (hasMixins && nativeCssVariables && this._applyShim) {
+        this._applyShim['transformRules'](ast, elementName);
+      }
+      template['_styleAst'] = ast;
     }
-    template['_styleAst'] = ast;
-
     let ownPropertyNames = [];
     if (!nativeCssVariables) {
       ownPropertyNames = StyleProperties.decorateStyles(template['_styleAst']);
     }
     if (!ownPropertyNames.length || nativeCssVariables) {
       let root = nativeShadow ? template.content : null;
       let placeholder = getStylePlaceholder(elementName);
-      let style = this._generateStaticStyle(info, template['_styleAst'], root, placeholder, cssBuild);
+      let style = this._generateStaticStyle(info, template['_styleAst'], root, placeholder, cssBuild, optimalBuild ? cssText : '');
```

Comment:
Can we refactor this to avoid generating a new style and instead just move the one from the template in optimalBuild case?